File: frontend/src/models/Spot.ts
import {
    BeachType,
    Discipline,
    ExperiencesLevel,
    Hazard,
    Month,
    ParkingSpace,
    WaterTemperature,
    WaveType,
    WindDirection
} from "./SpotSubTypes";
import {Position} from "./Position";

export type Spot= {
    id: string;
    name: string;
    disciplines: Discipline[];
    waveTypes: WaveType[];
    beachTypes: BeachType[];
    experiencesLevel: ExperiencesLevel[];
    hazards: Hazard[];
    bestMonths: Month[];
    bestDirections: WindDirection[];
    waterTemperature: WaterTemperature[];
    parkingSpace: ParkingSpace;
    position: Position;
    restrooms: string;
    spotImage?: FileList;
    imageBase64Encoded?: string
}

File: backend/src/main/java/com/github/nickklock/backend/models/spot/Spot.java
package com.github.nickklock.backend.models.spot;

import com.github.nickklock.backend.models.enums.*;
import com.github.nickklock.backend.models.geopostion.Position;
import com.mongodb.lang.Nullable;
import lombok.With;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@With
@Document("spots")
public record Spot(
        @Id
        String id,
        String name,
        List<Discipline> disciplines,
        List<WaveType> waveTypes,
        List<BeachType> beachTypes,
        List<ExperiencesLevel> experiencesLevel,
        List<Hazard> hazards,
        List<Month> bestMonths,
        List<WindDirection> bestDirections,
        List<WaterTemperature> waterTemperature,
        ParkingSpace parkingSpace,
        Position position,
        String restrooms,
        @Nullable
        String imageBase64Encoded
) {
}
